cmd/snap/cmd_run.go

```diff
@@ -1210,5 +1220,20 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	}
 }
 
+func getSnapDirOptions() (*dirs.SnapDirOptions, error) {
+	var opts dirs.SnapDirOptions
+
+	_, err := os.Stat(features.HiddenSnapDataHomeDir.ControlFile())
```

Comment:
isn't this SnapdFeature.IsEnabled ? (sorry for not mentioning this earlier)

Reply:
did you mention this? anyway if anything else it would mean we should fix IsEnabled separately, not starting ignoring it and working around it here and there. Even more generally we use FileExists quite extensively so if we think it's a nuisance we should address that but again separately.

Reply:
I didn't think FileExists was broken, the behaviour is documented and can be ok, depending on the caller's expectations. I'll change this to use IsEnabled and address the issue with its implementation in a separate PR